SpatialGDK/Source/SpatialGDK/Private/EngineClasses/RPCContainer.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/RPCPayload.h"
+#include "SpatialConstants.h"
+
+#include "CoreMinimal.h"
+
+// TODO(Alex): remove this logic when singletons can be referenced without entity IDs (UNR-1456) - deprecated???
+struct FPendingRPCParams
+{
+	FPendingRPCParams(UObject* InTargetObject, UFunction* InFunction, SpatialGDK::RPCPayload&& InPayload, int InReliableRPCIndex = 0);
+	~FPendingRPCParams() = default;
+
+	TWeakObjectPtr<UObject> TargetObject;
+	UFunction* Function;
+
+	int ReliableRPCIndex;
+	SpatialGDK::RPCPayload Payload;
+};
+
+using FPendingRPCParamsPtr = TSharedPtr<FPendingRPCParams>;
+
+DECLARE_DELEGATE_RetVal_OneParam(bool, FProcessRPCDelegate, FPendingRPCParamsPtr)
```

Comment:
Mildly prefer these kind of things at the top of the file.

Reply:
They are slightly lower since the follow the definition they rely on.

Reply:
Moved up.